### DISCUSSION TITLE
Error 401 with message Unauthorized

### DISCUSSION THREAD

**Discussion by virginiavannini** [fattureincloud/api#85] [Q&A] [answered]
Hi, 
I followed the Google Sheets Apps Script Quickstart tutorial for invoice creation but when I try to run it, clicking on the create button, I get the following message with 401 code.

La richiesta non riuscita per https://api-v2.fattureincloud.it ha restituito il codice 401. Risposta del server troncata: {"error":{"message":"Unauthorized."}} (usa l'opzione muteHttpExceptions per esaminare la risposta completa)

I'm using the Bearer Authorization formatted as written in the tutorial (the word Bearer followed by a space and a Token String).

Any suggestion?

**Comment by emajo** [comment, marked as answer]
Hi, @virginiavannini if the token is set correctly the thing that could cause a 401 is trying to access the wrong company data, are you sure you are setting your correct company id in the url? the url should be like `https://api-v2.fattureincloud.it/c/{company_id}/issued_documents`, ex. `https://api-v2.fattureincloud.it/c/12345/issued_documents`

**Reply by virginiavannini** [reply]
I need my ID and I get it from GET /user/info (https://developers.fattureincloud.it/api-reference#tag--User)
In the example the ID given when use this method is 12345 but my ID has 7 digts.

There is another way to take this ID? Where I can find it?

**Reply by valmoz** [reply]
Hi Virginia,
to get the Company ID you need to use the [GET /user/companies](https://developers.fattureincloud.it/api-reference#get-/user/companies) method, as explained [here](https://developers.fattureincloud.it/docs/basics/company-scoped-methods)

**Reply by valmoz** [reply]
Hi @virginiavannini let us know id you resolved this issue.

Thanks
Mauro

**Reply by virginiavannini** [reply]
Yes, thank you!
